### ISSUE TITLE
`q-expansion-item` leaks memory if its child is not a `q-item-section`

### ISSUE THREAD

**Issue by FelixNumworks** [quasarframework/quasar#17405]
### What happened?

When I update the model of the default slot of `q-extension-item`, memory is leaked if this slot is not filled with a `q-item-section`

#### This leaks memory when `myString` is updated
```vue
  <q-expansion-item>
    <template #default>
      <div> <!-- I'm using a div here -->
        <q-input v-model="myString"></q-input>
      </div>
    </template>
  </q-expansion-item>
```

#### This works well when `myString` is updated
```vue
  <q-expansion-item>
    <template #default>
      <q-item-section> <!-- I'm using a q-item-section here -->
        <q-input v-model="myString"></q-input>
      </q-item-section>
    </template>
  </q-expansion-item>
```

### What did you expect to happen?

No memory leak

### Reproduction URL

https://stackblitz.com/edit/quasarframework-ygojpg?file=src%2Fpages%2FIndexPage.vue

### How to reproduce?

1. Go to the reproduction URL
2. Copy the project locally, (or just the index component in any quasar project)
3. Launch it and watch the memory go up through dev tools.

_I didn't manage to see the memory go up directly on stackblitz, but it might be due to the fact that the code is executed in an iframe_

### Flavour

Quasar CLI with Vite (@quasar/cli | @quasar/app-vite)

### Areas

Components (quasar)

### Platforms/Browsers

Chrome

### Quasar info output

```Shell
Operating System - Darwin(23.4.0) - darwin/arm64
NodeJs - 20.11.1

Global packages
  NPM - 10.7.0
  yarn - Not installed
  pnpm - Not installed
  bun - Not installed
  @quasar/cli - 2.4.1
  @quasar/icongenie - 4.0.0
  cordova - Not installed

Important local packages
  quasar - 2.16.4 -- Build high-performance VueJS user interfaces (SPA, PWA, SSR, Mobile and Desktop) in record time
  @quasar/app-vite - 2.0.0-beta.14 -- Quasar Framework App CLI with Vite
  @quasar/extras - 1.16.11 -- Quasar Framework fonts, icons and animations
  eslint-plugin-quasar - Not installed
  vue - 3.4.31 -- The progressive JavaScript framework for building modern web UI.
  vue-router - 4.4.0
  pinia - 2.1.7 -- Intuitive, type safe and flexible Store for Vue
  vuex - Not installed
  vite - 5.3.2 -- Native-ESM powered web dev build tool
  vite-plugin-checker - Not installed
  eslint - 8.57.0 -- An AST-based pattern checker for JavaScript.
  esbuild - 0.20.2 -- An extremely fast JavaScript and CSS bundler and minifier.
  typescript - 4.9.5 -- TypeScript is a language for application scale JavaScript development
  workbox-build - Not installed
  register-service-worker - Not installed
  electron - 31.1.0 -- Build cross platform desktop apps with JavaScript, HTML, and CSS
  electron-packager - Not installed
  @electron/packager - Not installed
  electron-builder - 24.13.3 -- A complete solution to package and build a ready for distribution Electron app for MacOS, Windows and Linux with “auto update” support out of the box
  @capacitor/core - Not installed
  @capacitor/cli - Not installed
  @capacitor/android - Not installed
  @capacitor/ios - Not installed

Quasar App Extensions
  @quasar/quasar-app-extension-testing-e2e-cypress - 6.1.0 -- A Quasar App Extension for Cypress e2e
  @quasar/quasar-app-extension-testing-unit-vitest - 1.0.0 -- A Quasar App Extension for running tests with Vitest
```

**Comment by rstoenescu** [comment]
If this has to do with anything, it would be with how Vue manages the memory, not with Quasar UI itself, but will investigate.
Can you reproduce with anything else other than a QInput though?

**Comment by FelixNumworks** [comment]
Yes the QInput part isn't relevant. I just used it in my example for simplicity. In reality I use a custom component with a more complex model.

**Comment by FelixNumworks** [comment]
If it can help you, I have hundreds of these large `__vnode` and `transition` items.

<img width="1210" alt="Screenshot 2024-07-31 at 16 25 46" src="https://github.com/user-attachments/assets/f43552bc-6252-401a-8b01-a52f7c1ecf7a">

<img width="1210" alt="Screenshot 2024-07-31 at 16 26 49" src="https://github.com/user-attachments/assets/af0708f1-ae5f-4f65-9718-ecaedcb24632">

**Comment by rstoenescu** [comment]
Investigated this with an automated vnode-retention test: a WeakRef marker attached to each render generation's slot vnode, 2000 `v-model` updates driven inside the expansion content, then forced GC in headless Chrome, comparing a plain `div` child vs `q-item-section` vs a control rendering the same content without `q-expansion-item`.

Results:

- On vue 3.4.31 (the version in your repro), every render generation is retained while the component stays mounted: 2001 of 2001 markers alive. It leaks **identically** with a plain `div` and with `q-item-section`, so the child wrapper makes no difference. The control without `q-expansion-item` does not leak, which points at the `<Transition>` that QSlideTransition wraps the expansion content in.
- This is the upstream Vue bug vuejs/core#12181, fixed by vuejs/core#12182 and released in **vue 3.5.13**. A version bisect confirms it: leaks up to and including 3.5.12, clean from 3.5.13 onward.
- On current vue (3.5.41), exactly one vnode generation stays alive while mounted and everything is released on unmount, in both prod and dev builds, for both markup variants.

So the leak was in Vue core, not in Quasar. Upgrading to `vue >= 3.5.13` resolves it. Closing.
